rxbinding/src/main/java/com/jakewharton/rxbinding/view/ViewSystemUiVisibilityChangeEventOnSubscribe.java

```diff
@@ -0,0 +1,37 @@
+package com.jakewharton.rxbinding.view;
+
+import android.view.View;
+import com.jakewharton.rxbinding.internal.MainThreadSubscription;
+import rx.Observable;
+import rx.Subscriber;
+
+import static com.jakewharton.rxbinding.internal.Preconditions.checkUiThread;
+
+final class ViewSystemUiVisibilityChangeEventOnSubscribe
+    implements Observable.OnSubscribe<ViewSystemUiVisibilityChangeEvent> {
+  private final View view;
+
+  ViewSystemUiVisibilityChangeEventOnSubscribe(View view) {
+    this.view = view;
+  }
+
+  @Override public void call(final Subscriber<? super ViewSystemUiVisibilityChangeEvent> subscriber) {
+    checkUiThread();
+
+    final View.OnSystemUiVisibilityChangeListener listener =
+        new View.OnSystemUiVisibilityChangeListener() {
+          @Override public void onSystemUiVisibilityChange(int visibility) {
+            if (!subscriber.isUnsubscribed()) {
+              subscriber.onNext(ViewSystemUiVisibilityChangeEvent.create(view, visibility));
+            }
+          }
+        };
+    view.setOnSystemUiVisibilityChangeListener(listener);
+
+    subscriber.add(new MainThreadSubscription() {
+      @Override protected void onUnsubscribe() {
+        view.setOnTouchListener(null);
```

Comment:
Yes, didn't notice (autocomplete) -.- will fix it when I'm back.. thanks
for noticing
On Sep 2, 2015 5:10 PM, "Matthew Precious" <notifications@github.com> wrote:

> In
> rxbinding/src/main/java/com/jakewharton/rxbinding/view/ViewSystemUiVisibilityChangeEventOnSubscribe.java
> <https://github.com/JakeWharton/RxBinding/pull/118#discussion_r38543451>:
>
> > +  @Override public void call(final Subscriber<? super ViewSystemUiVisibilityChangeEvent> subscriber) {
> > +    checkUiThread();
> > +
> > +    final View.OnSystemUiVisibilityChangeListener listener =
> > +        new View.OnSystemUiVisibilityChangeListener() {
> > +          @Override public void onSystemUiVisibilityChange(int visibility) {
> > +            if (!subscriber.isUnsubscribed()) {
> > +              subscriber.onNext(ViewSystemUiVisibilityChangeEvent.create(view, visibility));
> > +            }
> > +          }
> > +        };
> > +    view.setOnSystemUiVisibilityChangeListener(listener);
> > +
> > +    subscriber.add(new MainThreadSubscription() {
> > +      @Override protected void onUnsubscribe() {
> > +        view.setOnTouchListener(null);
>
> Did you mean setOnSystemUiVisibilityChangeListener?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JakeWharton/RxBinding/pull/118/files#r38543451>.
>